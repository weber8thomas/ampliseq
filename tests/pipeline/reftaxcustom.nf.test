nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_reftaxcustom"
    tag "dada2"
    tag "pipeline"

    test("Custom DADA2 Reference Taxonomy Database") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/software_versions.yml").exists() },
                { assert snapshot(path("$outputDir/overall_summary.tsv")).match("overall_summary_tsv") },
                { assert snapshot(path("$outputDir/barrnap/rrna.arc.gff"),
                                path("$outputDir/barrnap/rrna.bac.gff"),
                                path("$outputDir/barrnap/rrna.euk.gff"),
                                path("$outputDir/barrnap/rrna.mito.gff")).match("barrnap") },
                { assert new File("$outputDir/barrnap/summary.tsv").exists() },
                { assert snapshot(path("$outputDir/cutadapt/cutadapt_summary.tsv")).match("cutadapt") },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                path("$outputDir/dada2/ASV_table.tsv"),
                                path("$outputDir/dada2/DADA2_stats.tsv"),
                                path("$outputDir/dada2/DADA2_table.rds"),
                                path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert new File("$outputDir/dada2/ASV_tax.user.tsv").exists() },
                { assert new File("$outputDir/dada2/ASV_tax_species.user.tsv").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1a_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_1a_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2a_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/sampleID_2a_2_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/input/Samplesheet.tsv")).match("input") },
                { assert snapshot(path("$outputDir/kraken2/ASV_tax.user.kraken2.classifiedreads.txt"),
                                path("$outputDir/kraken2/ASV_tax.user.kraken2.complete.tsv"),
                                path("$outputDir/kraken2/ASV_tax.user.kraken2.tsv")).match("kraken2") },
                { assert new File("$outputDir/qiime2/taxonomy/taxonomy.tsv").exists() },
                { assert new File("$outputDir/qiime2/taxonomy/GTGYCAGCMGCCGCGGTAA-GGACTACNVGGGTWTCTAAT-classifier.qza").exists() },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_cutadapt.txt")).match("multiqc") },
                { assert new File("$outputDir/summary_report/summary_report.html").exists() },
                { assert new File("$outputDir/phyloseq/dada2_phyloseq.rds").exists() },
                { assert new File("$outputDir/phyloseq/kraken2_phyloseq.rds").exists() }
            )
        }
    }
}
