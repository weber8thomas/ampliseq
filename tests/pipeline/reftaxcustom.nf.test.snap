{
    "input": {
        "content": [
            "Samplesheet.tsv:md5,dbf8d1a2b7933dab9e5a139f33c2b1f4"
        ],
        "timestamp": "2023-05-28T21:18:54+0000"
    },
    "cutadapt": {
        "content": [
            "cutadapt_summary.tsv:md5,5d02749984a811479e7d534fda75163f"
        ],
        "timestamp": "2023-05-28T21:18:54+0000"
    },
    "overall_summary_tsv": {
        "content": [
            "overall_summary.tsv:md5,3231d6ee72b9a1e7742e5605caaff05a"
        ],
        "timestamp": "2023-05-28T21:18:54+0000"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,864c3e0dc9b4a7649beee0c8665dceb5",
            "ASV_table.tsv:md5,2618251e597593e4d716dd9bed095539",
            "DADA2_stats.tsv:md5,54a1ac8d6c5a3ff15f700c4b2dd40c86",
            "DADA2_table.rds:md5,d095501019ce7ebccfa0eb801db1ed29",
            "DADA2_table.tsv:md5,5c9fb0bfd70da165f0ce6a361bfe0b43"
        ],
        "timestamp": "2023-05-28T21:18:54+0000"
    },
    "barrnap": {
        "content": [
            "rrna.arc.gff:md5,6dae470aace9293d5eb8c318584852dd",
            "rrna.bac.gff:md5,439a9084f089120f700f938dfb58fa41",
            "rrna.euk.gff:md5,c9bc1d9d8fb77dc19c95dee2d53840eb",
            "rrna.mito.gff:md5,df19e1b84ba6f691d20c72b397c88abf"
        ],
        "timestamp": "2023-05-28T21:18:54+0000"
    },
    "kraken2": {
        "content": [
            "ASV_tax.user.kraken2.classifiedreads.txt:md5,8a4693c37d5c24b342ef161b92567764",
            "ASV_tax.user.kraken2.complete.tsv:md5,3613dac9ce1bf03f87b57d1523e705f1",
            "ASV_tax.user.kraken2.tsv:md5,95c3f9daa5da8fe00159fb07d394c3ce"
        ],
        "timestamp": "2023-09-15T21:16:26+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_fastqc.txt:md5,147764e40079c3abf97a17cfe2275c52",
            "multiqc_general_stats.txt:md5,5cdf44a2a9fb47a9d847eaa68fb770b7",
            "multiqc_cutadapt.txt:md5,b4409890ffb9c6938433c374b50c380e"
        ],
        "timestamp": "2023-05-28T21:18:54+0000"
    }
}
